lib/iris/fileformats/netcdf.py

```diff
@@ -1299,6 +1299,14 @@ def _cf_coord_identity(self, coord):
             elif coord.standard_name == "longitude":
                 units = 'degrees_east'
 
+        elif isinstance(coord.coord_system, iris.coord_systems.RotatedGeogCS):
+            units = 'degrees'
+
+        elif isinstance(coord.coord_system,
+                        (iris.coord_systems.TransverseMercator,
+                         iris.coord_systems.LambertAzimuthalEqualArea)):
+            units = 'm'
+
```

Comment:
Right, I think I must have gotten confused trying to rebase a PR that was over a year old!

On closer inspection I now see that, in the case of tranverse_mercator we're just passing the units from the coord. Although this has it's disadvantages (i.e. it means that a user could produce a netcdf file with a badly defined coordinate:
e.g. `DimCoord(array([30, 60]), standard_name='latitude', units=Unit('m'), coord_system=RotatedGeogCS(10.0, 20.0))`
But I would prefer for users to have this flexibility, and cf says that there is no default value for lat or lon units.

As the CF conventions recommend a specific unit for GeogCS and RotatedGeogCS I think you could argue that we should 'enforce' these:
>The recommended unit of latitude is degrees_north

and
>Coordinates of latitude with respect to a rotated pole should be given units of degrees